docs/apache-airflow-providers-google/deprecation-policy.rst

```diff
@@ -0,0 +1,60 @@
+
+ .. Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+ ..   http://www.apache.org/licenses/LICENSE-2.0
+
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+ ..  NOTE! THIS FILE IS AUTOMATICALLY GENERATED AND WILL BE
+     OVERWRITTEN WHEN PREPARING PACKAGES.
+
+ ..  IF YOU WANT TO MODIFY THIS FILE, YOU SHOULD MODIFY THE TEMPLATE
+     `PROVIDER_CHANGELOG_TEMPLATE.rst.jinja2` IN the `dev/breeze/src/airflow_breeze/templates` DIRECTORY
+
+Deprecation policy for apache-airflow-providers-google package
+----------------------------------------------------------------
+
+Versioning of the package
+`````````````````````````
+
+As mentioned in `Airflow's release process and version policy <https://airflow.apache.org/docs/apache-airflow/stable/release-process.html#airflow-s-release-process-and-version-policy>`__
+Google provider package (and others) should follow SemVer, meaning that any breaking changes should be released together with bumping major version of the package.
+The change is considered to be a breaking if a DAG that was working before stops to work after the change.
+
+Deprecation Procedure
+`````````````````````
+
+The entire procedure of deprecating (either method, parameter or operator) consists of two steps:
+  - Using decorator to mark an object as deprecated, specifying planned removal date and an object that can be used instead of deprecated one:
+
+  .. code-block:: bash
+
+      from airflow.providers.google.common.deprecated import deprecated
+
+      @deprecated(
+          planned_removal_date="September 30, 2025",
+          use_instead="airflow.providers.google.cloud.hooks.vertex_ai.auto_ml.AutoMLHook, "
+          category=AirflowProviderDeprecationWarning,
+      )
+
+
+  - Once the date of the deprecated method/parameter/operator is passed, it can be removed (together with bumping major version of the package)
```

Comment:
Instead of writing inside of parentheses, what about writing in a new bullet point to write something like this:
```
- Since removing the method/parameter/operator is breaking the backward compatibility, we need to bump the major version.
```
I know we already mentioned about breaking changes, but still 😅

Reply:
I just removed the brackets :)